mailbox/store/src/main/java/org/apache/james/mailbox/store/StoreMailboxManager.java

```diff
@@ -521,7 +521,7 @@ private void validateDestinationPath(MailboxPath newMailboxPath, MailboxSession
     }
 
     private void assertIsOwner(MailboxSession mailboxSession, MailboxPath mailboxPath) throws MailboxNotFoundException {
-        if (!mailboxPath.belongsTo(mailboxSession)) {
+        if (mailboxSession.getType() != MailboxSession.SessionType.System && !mailboxPath.belongsTo(mailboxSession)) {
```

Comment:
This is the same condition than for create.

Please extract it to a well named sub-method (sessionHasAdministrativeAccessOn ?)

Reply:
Idem naming `assertIsOwner` is now outdated.